Start opendaylight as systemd service and add option to modify web port

nvin-krja opened the pull request:
Karaf container runs as java application and we cannot detect if
karaf is already running if there is any other active java process.
opendaylight rpm package will include .service file for controller.

Changed the logfile place holder for barclamp and replaced it with
a list of features to be installed in opendaylight controller.

I wanted to keep the logfile parameter but as opendaylight runs as
a service, to keep logfile parameter I must add KARAF_REDIRECT environment
variable in opendaylight.service file or add a new .conf file
for opendaylight and include that in opendaylight.service file. Too
much hassle for a single parameter. Will do that in future if we
decide to add more configurable parameter for karaf.

Commit:
Start opendaylight as systemd service

Karaf container runs as java application and we cannot detect if
karaf is already running if there is any other active java process.
opendaylight rpm package will include .service file for controller.

Changed the logfile place holder for barclamp and replaced it with
a list of features to be installed in opendaylight controller.

I wanted to keep the logfile parameter but as opendaylight runs as
a service, to keep logfile parameter I must add KARAF_REDIRECT environment
variable in opendaylight.service file or add a new .conf file
for opendaylight and include that in opendaylight.service file. Too
much hassle for a single parameter. Will do that in future if we
decide to add more configurable parameter for karaf.

hwoarang commented on chef/cookbooks/opendaylight/recipes/default.rb:
```diff
@@ -19,22 +19,21 @@
-         ;;
-      esac
-    EOF
+    retries 3
```
Why 3? What happens with the default 0 value? if there something fishy going on here could you add a comment so we remember in the future?

nvin-krja replied:
Karaf takes some time to start ssh access. As this runs immediately after odl service start, it fails if I don't wait. I don't know how much time it takes. So three retries with 2 seconds granularity.

hwoarang replied:
no problem but please add a comment to explain the situation so we will not forget and remove this part of code in the future.

hwoarang commented on chef/cookbooks/opendaylight/recipes/default.rb:
```diff
@@ -19,22 +19,21 @@
-      esac
-    EOF
+    retries 3
+    retry_delay 2
```
same comment as above

mmnelemane commented on chef/cookbooks/opendaylight/attributes/default.rb:
```diff
@@ -17,4 +17,4 @@
 #
 
-default[:opendaylight][:logfile] = "/dev/null"
+default[:opendaylight][:features] = "odl-restconf odl-l2switch-switch odl-dlux-core odl-ovsdb-openstack"
```
It would be better to have subsections for features: (for eg: 

```
features: {
    restconf:
      enabled: true
    l2switch:
      switch: true
    dlux: 
       core: true
       sfc: false
   ovsdb:
      openstack: true
      southbound: false
}
```

This structure will help us add specific configurations for each feature set.
Imagine this line with 15+ features in one line.

nvin-krja replied:
It will look good if we do that but it will cause confusion. There are around  20-30 features in ODL and as we not doing anything additional other than installing them, we will have to include all of them  in subsections. For now this is ok i think. Later once we start fully supporting ODL features, we can add check box for each feature.

Commit:
Add option to change opendaylight web portal port

Opendaylight uses two ports for web portal - 8080 and 8181. 8080 is the
primary port and 8181 is the backup ADSAL port. Port 8080 conflicts with
swift-proxy port. Changing it to 8070 and adding option in barclamp to
set it.

mmnelemane commented:
There hasn't been any further comments on this since last week. IMO the changes are harmless.
